tail_reader.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"io"
+	"os"
+	"time"
+)
+
+type tailReader struct {
+	io.ReadCloser
+}
+
+// Read provides a tailf like generator.
+//
+// The reader sleeps on EOF and is shutdown when the file is closed.
+// Implements the io.Reader inerface.
+func (t tailReader) Read(b []byte) (int, error) {
+	for {
+		n, err := t.ReadCloser.Read(b)
+		if n > 0 {
+			return n, nil
+		} else if err != io.EOF {
+			return n, err
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
Can you elaborate on why is this sleep needed?

Reply:
Thanks for commenting on this. I added sleep here as a way to give up control to the scheduler while developing the feature. Nevertheless,  using it here feels like an anti-pattern and doesn't add anything so I removed it.